library/cartridge_edit_topology.py

```diff
@@ -366,25 +463,39 @@ def edit_topology(params):
             return helpers.ModuleRes(failed=True, msg="Failed to edit topology: %s" % err)
 
         topology_changed = True
-        edited_replicasets = res['replicasets']
-        edited_instances = res['servers']
-
-        # update replicasets
-        for alias, replicaset in edited_replicasets.items():
-            cluster_replicasets[alias] = replicaset
-
-        # update instances
-        for alias, instance in edited_instances.items():
-            cluster_instances[alias] = instance
+        update_instances_and_replicasets(res, cluster_replicasets, cluster_instances)
 
     # change failover priority if needed
-    edit_topology_params = get_edit_failover_priority_params(replicasets, cluster_replicasets, cluster_instances)
+    edit_topology_params, err = get_edit_failover_priority_params(replicasets, cluster_replicasets, cluster_instances)
+    if err is not None:
+        return helpers.ModuleRes(
+            failed=True,
+            msg="Failed to collect edit topology params for changing failover_priority: %s" % err
+        )
+
     if edit_topology_params:
         res, err = control_console.eval_res_err(edit_topology_func_body, edit_topology_params)
         if err is not None:
             return helpers.ModuleRes(failed=True, msg="Failed to edit failover priority: %s" % err)
 
         topology_changed = True
+        update_instances_and_replicasets(res, cluster_replicasets, cluster_instances)
+
+    # configure instances
+    edit_topology_params, err = get_configure_instances_params(instances, cluster_instances)
+    if err is not None:
+        return helpers.ModuleRes(
+            failed=True,
+            msg="Failed to collect edit topology params for instances: %s" % err
```

Comment:
Let's make message more accurate:
```suggestion
            msg="Failed to collect edit topology params to configure instances: %s" % err
```